independent-projects/arc/runtime/src/main/java/io/quarkus/arc/ActiveResult.java

```diff
@@ -0,0 +1,86 @@
+package io.quarkus.arc;
+
+import java.util.Objects;
+
+/**
+ * Result of the "is active?" question asked against synthetic beans.
+ * Can say whether the bean {@linkplain #result() is active} and if it is not,
+ * what is the {@linkplain #inactiveReason() reason}.
+ */
+public final class ActiveResult {
```

Comment:
Maybe use java record?

Reply:
Public records always have a public canonical constructor with all of their components, which is not what we want here.

Reply:
Why? You can keep the static factory methods and perform the checks in the constructor...

Reply:
Nevermind, it's just a nitpicking comment ;-).

Reply:
Sure, but why would we want a public constructor in addition to the factory methods?

Reply:
I don't see that as a problem. You would save 3 methods and 3 fields definitions and just add one more way to instantiate the object, with the same semantics :shrug:.

Reply:
I just don't like the public constructor, but if you insist, I can rewrite this :-)

Reply:
I do not insist at all.